platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -365,7 +375,24 @@ public void preHandle(@NonNull final Event event) {
      * {@inheritDoc}
      */
     @Override
-    public MerkleNode migrate(final int ignored) {
+    public MerkleNode migrate(final int version) {
+        if (version < VERSION_2) {
+            final var zerothChild = getChild(0);
+            if (!(zerothChild instanceof PlatformState platformState)) {
+                throw new IllegalStateException("Expected a PlatformState as the first child");
```

Comment:
it's worth having a unit test specifically for this method, which should include this condition. I see references in SerializationTest but i don't think they test the  unhappy paths like this or the case version isn't < VERSION_2 or if child == null on line 391.

Reply:
I added unit tests for this code